hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,295 @@
+        makeAllPicks(timestep, N_evap , N_bounced_total);
+        
+        /*
+         * Select the picks that lie on my rank, with reindexing to local mark indexes.
+         * This is performed in a do loop to allow for resizing of the GPUVector.
+         * After a short time, the loop will be ignored.
+         */
+
+        const unsigned int max_pick_idx = N_before + N_bounced;
+        bool overflowed = false;
+        do
+            {
+            m_Npick = 0;
+            const unsigned int max_Npick = m_picks.getNumElements();
+
+                {
+                ArrayHandle<unsigned int> h_picks(m_picks, access_location::host, access_mode::overwrite);
+                for (unsigned int i=0; i < N_evap; ++i)
+                    {
+                    const unsigned int pick = m_all_picks[i];
+                    if (pick >= N_before && pick < max_pick_idx)
+                        {
+                        if (m_Npick < max_Npick)
+                            {
+                            h_picks.data[m_Npick] = pick - N_before;
+                            }
+                        ++m_Npick;
+                        }
+                    }
+                }
+
+            overflowed = (m_Npick > max_Npick);
+            if (overflowed)
+                {
+                m_picks.resize(m_Npick);
+                }
+
+            } while (overflowed);
+        }
+    
+    applyPicks();
+    }
+
+void DryingDropletStreamingMethod::makeAllPicks(unsigned int timestep, unsigned int N_pick, unsigned int N_bounced_total)
+    {
+    assert(N_pick <= N_bounced_total);
+
+    // fill up vector which we will randomly shuffle
+    m_all_picks.resize(N_bounced_total);
+    std::iota(m_all_picks.begin(), m_all_picks.end(), 0);
+
+    hoomd::RandomGenerator rng(hoomd::RNGIdentifier::DryingDropletStreamingMethod, m_seed, timestep);
+
+    // random shuffle (fisher-yates) to get picks, seeded the same across all ranks
+    auto begin = m_all_picks.begin();
+    auto end = m_all_picks.end();
+    size_t left = std::distance(begin,end);
+    unsigned int N_choose = N_pick;
+    while (N_choose-- && left > 1)
+        {
+        hoomd::UniformIntDistribution rand_shift(left-1);
+
+        auto r = begin;
+        std::advance(r, rand_shift(rng));
+        std::swap(*begin, *r);
+        ++begin;
+        --left;
+        }
+
+    // size the vector down to the number picked
+    m_all_picks.resize(N_pick);
+    }
+
+void DryingDropletStreamingMethod::applyPicks()
+    {
+    ArrayHandle<unsigned int> h_picks(m_picks, access_location::host, access_mode::read);
+    ArrayHandle<unsigned int> h_bounced_index(m_bounced_index, access_location::host, access_mode::read);
+    ArrayHandle<unsigned char> h_bounced(m_bounced, access_location::host, access_mode::readwrite);
+    for (unsigned int i=0; i < m_Npick; ++i)
+        {
+        const unsigned int pidx = h_bounced_index.data[h_picks.data[i]];
+        h_bounced.data[pidx]= 0b11;
```

Comment:
The canonical way to set a single bit is
```suggestion
        h_bounced.data[pidx] |= 1 << 1;
```
since the bit shift tells us which bit we're setting. Use of `|=` means you don't overwrite any of the existing bits.